src/Middleware/RateLimiting/src/RateLimitingServicesExtensions.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.RateLimiting;
+
+/// <summary>
+/// Extension methods for the RateLimiting middleware.
+/// </summary>
+public static class RateLimitingServicesExtensions
+{
+
+    /// <summary>
+    /// Adds rate limiting services.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> for adding services.</param>
+    /// <param name="configureOptions">A delegate to configure the <see cref="RateLimitingOptions"/>.</param>
+    /// <returns></returns>
+    public static IServiceCollection AddRateLimiting(this IServiceCollection services, Action<RateLimitingOptions> configureOptions)
```

Comment:
Should this be called `ConfigureRateLimiting` since that is all it does?

Reply:
We need a `UseRateLimiting` extension method on `IApplicationBuilder`.

I think this is a lot like `UseStaticFiles` where if someone wants to call `services.Configure<RateLimitingOptions>(options => ...` they're free to do so, but most people will just pass the options to a `UseRateLimiting(RateLimitingOptions)` overload. I don't think `AddStaticFiles` exists.